proxy/coco/src/peer/subroutines.rs

```diff
@@ -0,0 +1,280 @@
+//! Management of subroutines.
```

Comment:
Could you expand this doc? What problems does it address and what should/should NOT go in here would be nice to add